### ISSUE TITLE
Reevaluating npm workspaces

### ISSUE THREAD

**Issue by egekorkan** [eclipse-thingweb/playground#379]
Following up on https://github.com/eclipse/thingweb.node-wot/pull/483, we can reevaluate whether npm workspaces make more sense for us.

This can help with the #366

**Comment by hasanheroglu** [comment]
https://tyhopp.com/notes/replacing-lerna I've found this, which explains the advantages & disadvantages switching from lerna to workspaces. To sum it up:
- `npm install` works fine also with workspaces, even easier since we run for all packages at once.
- Versioning needs additional scripting if we want to bump versions of all the packages. (I'm not sure if it is relevant bumping all the packages at once for our repo) But even though we want to bump versions of all the packages we could handle that with a simple script.
- Creating changelog automatically is not possible with workspaces, but I don't know how useful it is since lerna includes all the changes that happened which could create changelogs with too much unnecessary information.

Apart from that, there are advantages of workspaces as mentioned on https://github.com/eclipse/thingweb.node-wot/pull/483, such as:
- workspaces included in npm
- via symlink less disk space is used
- only one package-lock.json
- it seems more possible for workspaces to be maintained longer than lerna, since npm is widely used

In conclusion, I think I could say we can try to switch to workspaces and see what happens.

**Comment by egekorkan** [comment]
Thanks @hasanheroglu for the analysis. I would then agree that npm workspaces makes more sense. For the record, bumping all packages at the same would happen in cases such as major TD version updates that mean major version change for all packages.

We should work on having a changelog as well, not incredibly soon though :)

**Comment by egekorkan** [comment]
This is irrelevant in the future since all tools but the web and maybe the cli will be moved to td-tools repository where no npm workspaces will be used
